src/pybind/ceph_daemon.py

```diff
@@ -43,9 +43,12 @@ def do_sockio(path, cmd_bytes):
             l, = struct.unpack(">I", len_str)
             sock_ret = b''
 
-            got = 0
+            got = long(0)
             while got < l:
-                bit = sock.recv(l - got)
+                # recv only accept signed int, i.e max 2GB
+                # walk around by limit each recv to 128M
+                want = min(l - got, 128*1024*1024)
```

Comment:
i'd suggest define a "relatively small power of 2, for example, 4096." [1], i.e.

```py
READ_CHUNK_SIZE = 4096
```

and use it instead,

---
[1] https://docs.python.org/2/library/socket.html